File: v1/Demo/source/demo_get_velocity.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
    usleep(1000000);

    vector<float> velocity = api.get_velocity();

    if (velocity.size() != 6)
    {
        Logger::get_instance()->print_trace_error("get velocity failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get velocity success : ";
    for (int i = 0; i < velocity.size(); ++i) {
        std::cout << velocity[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}

File: v1/Demo/demo_get_velocity.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    velocity = api.get_velocity()

    if len(velocity) != 6:
        logger.print_trace_error("get velocity failed")
    else:
        logger.print_trace("get velocity success", velocity)

if __name__ == "__main__":
    main()
